xarray/plot/plot.py

```diff
@@ -143,14 +172,20 @@ def plot(darray, row=None, col=None, col_wrap=None, ax=None, hue=None,
     """
     darray = darray.squeeze()
 
-    if contains_cftime_datetimes(darray):
-        raise NotImplementedError(
-            'Built-in plotting of arrays of cftime.datetime objects or arrays '
-            'indexed by cftime.datetime objects is currently not implemented '
-            'within xarray. A possible workaround is to use the '
-            'nc-time-axis package '
-            '(https://github.com/SciTools/nc-time-axis) to convert the dates '
-            'to a plottable type and plot your data directly with matplotlib.')
+    # If I am not mistaken, this did check only if there are cftime.datetime
+    # objects in the actual data, not dims.
+    # Correction below
+
+    if any([contains_cftime_datetimes(darray[dim]) for dim in darray.dims]):
+        if nc_axis_available:
+            darray = _convert_all_cftime(darray)
+        else:
+            raise ImportError(
+                'Built-in plotting of arrays of cftime.datetime objects or  '
+                'arrays indexed by cftime.datetime objects requires the '
+                'optional `nc-time-axis` package '
+                '(https://github.com/SciTools/nc-time-axis).'
+            )
```

Comment:
The old version of `contains_cftime_datetimes` was never triggered for my example. Not sure if this was a bug or if I am missing something here.

Reply:
Yes, thanks, checking the dimensions would definitely be an improvement 👍.  We probably still want to check that the data does not contain `cftime.datetime` objects too.

Reply:
Thinking about this a little more, technically, I think you *could* still pass a time dimension filled with cftime dates to the `row` or `col` argument of `plot`, and things would still work without nc-time-axis (because you would not be plotting cftime dates in that case; they would just be in the plot titles). 

For that reason, it might be slightly cleaner to move this check to `_ensure_plottable`, because that only gets called on things that xarray tries to plot.  Something like:
```python
def _ensure_plottable(*args):
    """
    Raise exception if there is anything in args that can't be plotted on an
    axis by matplotlib.
    """
    numpy_types = [np.floating, np.integer, np.timedelta64, np.datetime64]
    other_types = [datetime]

    try:
        import cftime
        cftime_datetime = [cftime.datetime]
    except ImportError:
        cftime_datetime = []
    other_types = other_types + cftime_datetime

    for x in args:
        if not (_valid_numpy_subdtype(np.array(x), numpy_types)
                or _valid_other_type(np.array(x), other_types)):
            raise TypeError('Plotting requires coordinates to be numeric '
                            'or dates of type np.datetime64, '
                            'datetime.datetime, or cftime.datetime or '
                            'pd.Interval.')
        if (_valid_other_type(np.array(x), cftime_datetime)
            and not nc_axis_available):
            raise ImportError('Plotting of arrays of cftime.datetime '
                              'objects or arrays indexed by cftime.datetime '
                              'objects requires the optional `nc-time-axis` '
                              'package.')
```

Reply:
I am super sorry @spencerkclark. I misread your example and then had to submit a paper. I am working on it now.

Reply:
All good @jbusecke -- I appreciate your help with this.